skiko/src/awtMain/cpp/windows/directXSwingRedrawer.cc

```diff
@@ -0,0 +1,400 @@
+#ifdef SK_DIRECT3D
+
+#include <locale>
+#include <Windows.h>
+#include <jawt_md.h>
+#include "jni_helpers.h"
+#include "exceptions_handler.h"
+#include "window_util.h"
+
+#include "SkColorSpace.h"
+#include "GrBackendSurface.h"
+#include "GrDirectContext.h"
+#include "SkSurface.h"
+#include "../common/interop.hh"
+
+#include "d3d/GrD3DTypes.h"
+#include <d3d12sdklayers.h>
+#include "d3d/GrD3DBackendContext.h"
+#include <d3d12.h>
+#include <dxgi1_4.h>
+#include <dxgi1_6.h>
+
+class DirectXOffscreenDevice
```

Comment:
does it matter?

Reply:
In case of class I'd not put everything to public and add methods to work with them (encapsulation, right?)

Reply:
You imply semantics which aren't declared somewhere in the project. Moreover, on one of previous PRs I was asked not to put any logic outside of JNI methods, because it makes it harder to navigate between JNI methods bodies and code (without a lot of tinkering).

Reply:
Writing this code properly (using smart pointers, encapsulating all within methods) would turn this into code I was specifically asked not to produce.